lib/puppet/network/formats.rb

```diff
@@ -101,17 +101,19 @@ def data_to_instance(klass, data)
 
 Puppet::Network::FormatHandler.create_serialized_formats(:json, :mime => 'application/json', :charset => Encoding::UTF_8, :weight => 15, :required_methods => [:render_method, :intern_method], :intern_method => :from_data_hash) do
   def intern(klass, text)
-    data_to_instance(klass, JSON.parse(text))
+    data_to_instance(klass, Puppet::Util::Json.load(text))
   end
 
   def intern_multiple(klass, text)
-    JSON.parse(text).collect do |data|
+    Puppet::Util::Json.load(text).collect do |data|
       data_to_instance(klass, data)
     end
   end
 
   # JSON monkey-patches Array, so this works.
   # https://github.com/ruby/ruby/blob/ruby_1_9_3/ext/json/generator/generator.c#L1416
+  # If we ever change what JSON backend we are using, we will need to make sure this is
+  # still true.
```

Comment:
This appears to not work right in Jackson/JrJackson, I get
```
Error: Could not send report: Error 500 on SERVER: Internal Server Error: org.jruby.exceptions.RaiseException: (NotImplementedError) Array does not respond to to_json; can not render instances to application/json
```
when attepting to do an agent run against a server with the new gems installed.